crates/wasm_interp/src/tests/test_mem.rs

```diff
@@ -49,6 +49,72 @@ fn test_growmemory() {
     assert_eq!(state.memory.len(), 5 * MemorySection::PAGE_SIZE as usize);
 }
 
+#[test]
+fn test_memory_fill() {
+    let arena = Bump::new();
+    let mut module = WasmModule::new(&arena);
+
+    let pages = 3;
+    let pc = 0;
+    module.memory = MemorySection::new(&arena, pages * MemorySection::PAGE_SIZE);
+
+    const SIZE: i32 = 16;
+    let byte_value = 0xAA;
+    let destination = 0x4;
+
+    let bytes = [OpCode::MEMORY as u8, 11, 0x0];
+    module.code.bytes.extend(bytes);
+
+    let mut state = Instance::new(&arena, pages, pc, [], DefaultImportDispatcher::default());
+
+    state.value_store.push(Value::I32(destination));
+    state.value_store.push(Value::I32(byte_value));
+    state.value_store.push(Value::I32(SIZE));
+
+    // before the instruction, the memory is all zeros
+    let slice = &state.memory[destination as usize..][..SIZE as usize];
+    assert_eq!(slice, &[0; SIZE as usize]);
+
+    state.execute_next_instruction(&module).unwrap();
+
+    let slice = &state.memory[destination as usize..][..SIZE as usize];
+    assert_eq!(slice, &[byte_value as u8; SIZE as usize])
+}
+
+#[test]
+fn test_memory_copy() {
+    let arena = Bump::new();
+    let mut module = WasmModule::new(&arena);
+
+    let pages = 3;
+    let pc = 0;
+    module.memory = MemorySection::new(&arena, pages * MemorySection::PAGE_SIZE);
+
+    const SIZE: i32 = 4;
+    let source = 0x4;
+    let destination = 0x8;
+
+    let bytes = [OpCode::MEMORY as u8, 10, 0x0, 0x0];
+    module.code.bytes.extend(bytes);
+
+    let mut state = Instance::new(&arena, pages, pc, [], DefaultImportDispatcher::default());
+
+    state.value_store.push(Value::I32(destination));
+    state.value_store.push(Value::I32(source));
+    state.value_store.push(Value::I32(SIZE));
+
+    // before the instruction, the memory is all zeros
```

Comment:
This comment seems misleading. The memory is not all zeros before the instruction, it contains `0xAA`.

Reply:
Ah I eventually figured out what's really going on. I think it would be helpful to make the comment more accurate, maybe mentioning destination memory or something like that.

I know these are nitpicky details but they are the exact nitpicky details that are being tested, so I think it matters!

Reply:
yes, this is why we do code review. I've clarified the comments.